pkg/api/api.go

```diff
@@ -189,7 +189,15 @@ func (a *API) RegisterRuntimeConfig(runtimeConfigHandler http.HandlerFunc) {
 func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distributor.Config) {
 	client.RegisterPushOnlyIngesterServer(a.server.GRPC, d)
 
-	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig, a.sourceIPs, d.Push), true, "POST")
+
+	var pushFunc push.Func
+	if pushConfig.DistributorPushWrapper != nil {
+		pushFunc = pushConfig.DistributorPushWrapper(d.Push)
+	} else {
+		pushFunc = d.Push
+	}
```

Comment:
How can I test this change? Should I make it part of the `distributor.Config`?

```golang
func (c *Config) Push(d *Distributor) push.Func {
    if pushConfig.DistributorPushWrapper != nil {
        return c.DistributorPushWrapper(d.Push)
    } else {
        return d.Push
    }
}
```

Reply:
I wouldn't worry about it too much, it's pretty straightforward logic.